deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -305,120 +306,123 @@ def running_simulations(self) -> Dict[str, List[str]]:
                         switches.append(line_stripped)
         return {'switches': switches, 'simdrivers': simdrivers}
 
-    def monitor_jobs_instance(self, completed_jobs: List[str], teardown: bool, terminateoncompletion: bool,
+    def monitor_jobs_instance(self,
+            prior_completed_jobs: List[str],
+            is_final_loop: bool,
+            is_networked: bool,
+            terminateoncompletion: bool,
             job_results_dir: str) -> Dict[str, Dict[str, bool]]:
         """ Job monitoring for this host. """
-        # make a local copy of completed_jobs, so that we can update it
-        completed_jobs = list(completed_jobs)
+        self.instance_logger(f"Final loop?: {is_final_loop} Is networked?: {is_networked} Terminateoncomplete: {terminateoncompletion}", debug=True)
+        self.instance_logger(f"Prior completed jobs: {prior_completed_jobs}", debug=True)
+
+        def do_terminate():
+            if (not is_networked) or (is_networked and is_final_loop):
+                if terminateoncompletion:
+                    self.terminate_instance()
 
-        rootLogger.debug("completed jobs " + str(completed_jobs))
 
         if not self.instance_assigned_simulations() and self.instance_assigned_switches():
```

Comment:
This section of the code is the same except for the `do_terminate` section. However, when the `do_terminate` is called here `is_final_loop == True` so the `if terminateoncompletion:` is always reached which makes the statement the same as before.